app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt

```diff
@@ -291,11 +291,11 @@ class ValueStoreImpl(
 
     private fun deleteDataElementValue(field: String, optionUid: String): StoreResult {
         val option = d2.optionModule().options().uid(optionUid).blockingGet()
-        val possibleValues = arrayListOf(option.name()!!, option.code()!!)
+        val possibleValues = arrayListOf(option?.name(), option?.code()).filterNotNull()
         val valueRepository =
             d2.trackedEntityModule().trackedEntityDataValues().value(recordUid, field)
         return if (valueRepository.blockingExists() &&
-            possibleValues.contains(valueRepository.blockingGet().value())
+            possibleValues.contains(valueRepository.blockingGet()?.value())
```

Comment:
I think that this methods always returns false is option is null, so maybe you can return false after the line 293 if it is null.